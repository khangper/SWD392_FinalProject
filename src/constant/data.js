export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
