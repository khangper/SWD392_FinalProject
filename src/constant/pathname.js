export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
