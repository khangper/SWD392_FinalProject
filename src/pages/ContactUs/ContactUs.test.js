import React from "react";
import { render, screen } from "@testing-library/react";
import ContactUS from "../ContactUs/ContactUs";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("ContactUs", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the ContactUs component", async () => {
    const { container } = renderWithProviders(<ContactUS />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
