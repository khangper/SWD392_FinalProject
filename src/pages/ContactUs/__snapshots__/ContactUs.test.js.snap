// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs renders the ContactUs component 1`] = `
<div>
  <header
    class="header1"
  >
    <a
      href="/home"
    >
      <button
        class="header-link1"
      >
        Back To Cursus
      </button>
    </a>
    <a
      href="/home"
    >
      <img
        alt=""
        class="header-group-icon"
        src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
      />
    </a>
    <img
      alt=""
      class="header-avatar"
      src="profile-img.jpg"
    />
  </header>
  <div
    class="Contact_Main"
  >
    <div
      class="Contact_Title"
    >
      <div
        class="Contact_Title_Container"
      >
        <div
          class="Contact_Small_Title"
        >
          <div
            class="Contact_Small_Title_Left"
          >
            <a
              href="/home"
            >
              Home 
            </a>
            <p>
               / Contact Us
            </p>
          </div>
          <div
            class="Contact_Small_Title_Right"
          >
            <a
              href="/home"
            >
              Back to Home
            </a>
          </div>
        </div>
        <div
          class="Contact_Big_Title"
        >
          <h2>
            Contact Us
          </h2>
        </div>
      </div>
    </div>
    <div
      class="Contact_Body"
    >
      <div
        class="Contact_Body_Box"
      >
        <a
          class="Contact_link_step"
          href="/help"
        >
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/help_icon.svg"
          />
          <h4>
            Help Center
          </h4>
        </a>
      </div>
      <div
        class="Contact_Body_Box"
      >
        <a
          class="Contact_link_step"
          href="blog"
        >
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/blog_icon.svg"
          />
          <h4>
            Blog
          </h4>
        </a>
      </div>
      <div
        class="Contact_Body_Box"
      >
        <a
          class="Contact_link_step"
          href="career"
        >
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/career_icon.svg"
          />
          <h4>
            Careers
          </h4>
        </a>
      </div>
      <div
        class="Contact_Body_Box"
      >
        <a
          class="Contact_link_step"
          href="comingsoon"
        >
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/developer_icon.svg"
          />
          <h4>
            Developer Area
          </h4>
        </a>
      </div>
      <div
        class="Contact_Map"
      >
        <iframe
          height="450"
          loading="lazy"
          referrerpolicy="no-referrer-when-downgrade"
          src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3918.682512094123!2d106.8050999758388!3d10.835590658098013!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x3175272fb77e7199%3A0x98cb51d4483e1172!2sFPT%20Software%20F-Town%203!5e0!3m2!1sen!2s!4v1721522116112!5m2!1sen!2s"
          style="border: 0px;"
          width="600"
        />
      </div>
      <div
        class="Contact_Information"
      >
        <div
          class="Contact_Checkout_Title"
        >
          <h4>
            Contact Information
          </h4>
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/line.svg"
          />
        </div>
        <div
          class="Contact_Detail_Inf"
        >
          <div
            class="Contact_txt_cntct"
          >
            <span
              class="Contact_cntct_895"
            >
              <i
                class="uil uil-location-point"
              />
              Main Address :
            </span>
            <p>
              #1235 Sks Nagar St No. 8 Phase 3, Pakhowal Road, 141001, LDH, Punjab, India
            </p>
          </div>
          <div
            class="Contact_txt_cntct"
          >
            <span
              class="Contact_cntct_895"
            >
              <i
                class="uil uil-envelope"
              />
              Email Address :
            </span>
            <p>
              Info@edututsplus@gmail.com
            </p>
          </div>
          <div
            class="Contact_txt_cntct"
          >
            <span
              class="Contact_cntct_895"
            >
              <i
                class="uil uil-mobile-android-alt"
              />
              Phone Number:
            </span>
            <p>
              +911234567890, 01610000000
            </p>
          </div>
          <div
            class="Contact_Social"
          >
            <ul
              class="social-icons"
            >
              <a
                class="sc-fb"
                href="#"
              >
                <img
                  src="facebook.png"
                />
              </a>
              <a
                class="sc-tw"
                href="#"
              >
                <img
                  src="x.png"
                />
              </a>
              <a
                class="sc-ln"
                href="#"
              >
                <img
                  src="linkedin.png"
                />
              </a>
              <a
                class="sc-yt"
                href="#"
              >
                <img
                  src="youtube.png"
                />
              </a>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="footer footer-large-container"
  >
    <div
      class="footer-links"
    >
      <div
        class="footer-column"
      >
        <a
          class="link"
          href="/about-us"
        >
          About
        </a>
        <a
          class="link"
          href="/blog-single-view/:id"
        >
          Blog
        </a>
        <a
          class="link"
          href="/career"
        >
          Careers
        </a>
        <a
          class="link"
          href="/press"
        >
          Press
        </a>
      </div>
      <div
        class="footer-column"
      >
        <a
          class="link"
          href="/help"
        >
          Help
        </a>
        <a
          class="link"
          href="/coming-soon"
        >
          Advertise
        </a>
        <a
          class="link"
          href="/coming-soon"
        >
          Developers
        </a>
        <a
          class="link"
          href="/contactus"
        >
          Contact Us
        </a>
      </div>
      <div
        class="footer-column"
      >
        <a
          class="link"
          href="/termsofuse"
        >
          Copyright Policy
        </a>
        <a
          class="link"
          href="/termsofuse"
        >
          Terms
        </a>
        <a
          class="link"
          href="/termsofuse"
        >
          Privacy Policy
        </a>
        <a
          class="link"
          href="/sitemap"
        >
          Sitemap
        </a>
      </div>
      <div
        class="footer-column"
      >
        <button
          class="teach-button"
        >
          Teach on Cursus
        </button>
        <div
          class="language-selector"
        >
          <button>
            Language 
            <span>
              ▼
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      <div
        class="footer-copy"
      >
        <img
          alt=""
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo1.svg"
        />
        © 2024 
        <strong>
          Cursus
        </strong>
        . All Rights Reserved.
      </div>
      <div
        class="footer-social"
      >
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="facebook.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="x.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="google+.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="linkedin.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="insta.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="pinterest.png"
          />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
